packages/components/src/DateTimePickers/pickers/CalendarPicker/CalendarPicker.component.js

```diff
@@ -27,6 +36,8 @@ class CalendarPicker extends React.Component {
 				year: getYear(initialCalendarDate),
 			},
 			selectedDate,
+			startDate,
+			endDate,
```

Comment:
Do not set them in state, just pass the props in the jsx. Here you duplicate the value, and never update it. And it's only to passe them in the jsx, just pass the props